controllers/users_controller.js

```diff
@@ -3,133 +3,118 @@ const utils = require('../utils/util')
 const jwt = require('jwt-simple')
 const cfg = require('../config')
 
-module.exports.findAllUsers = (req, res) => {
-    user.findAllUsers().subscribe(
-        users =>
-            res.status(200).json({
-                error: false,
-                users: users
-            }),
-        err =>
-            res.status(404).json({
-                error: true,
-                users: []
-            })
-    )
+module.exports.findAllUsers = async (req, res) => {
+    try {
+        const users = await user.findAllUsers()
+        res.status(200).json({
+            error: false,
+            data: users
+        })
+    }catch (e) {
+        console.log(e);
+        return res.status(500).json({error: true});
+    }
 }
 
-module.exports.register = ({body: {username = null, password = null, email = null, name = null}}, res) => {
-    let errorMessage = '';
+module.exports.register = async ({body: {username = null, password = null, email = null, name = null}}, res) => {
+    let errors = '';
 
     if (utils.isEmpty(username)) {
-        errorMessage += 'Username is Null\n'
+        errors.push('Username is null')
     }
     if (utils.isEmpty(password)) {
-        errorMessage += 'Password is Null\n'
+        errors.push('Password is null')
     }
     if (utils.isEmpty(email)) {
-        errorMessage += 'Email is Null\n'
+        errors.push('Email is null')
     }
     if (utils.isEmpty(name)) {
-        errorMessage += 'Name is Null\n'
+        errors.push('Name is null')
     }
 
-    if (!utils.isEmpty(errorMessage)) {
+    if (!utils.isEmpty(errors)) {
         res.status(401).json({
             error: true,
-            data: errorMessage
+            data: errors
         })
     }
     else {
-        user.register(username, password, email, name).subscribe(
-            id =>
-                res.status(201).json({
-                    error: false,
-                    data: id
-                }),
-            err => {
-                let message = '';
-                //noinspection EqualityComparisonWithCoercionJS
-                if (err.code == 23505) {
-                    message = `${utils.capitalize(err.detail.match(/[a-zA-Z]+(?=\))/)[0])} already exists`;
-                }
-                res.status(500).json({
+        try {
+            const {id, datecreated} = await user.register(username, password, email, name)
+            res.status(201).json({
+                error: false,
+                data: {name, username, email, datecreated, id}
+            })
+        }catch (e) {
+            if(e.code === '23505'){
```

Comment:
[indentation hell] this part would particularly need reworking

Reply:
How you would you suggest I make it nicer? I could remove the `else` since I'm going `return res.status(409)`

Reply:
The idiom I mentionned previously (early return so no need for an `else`), as well as going away from the `try{} catch{}` structure, but I understand you like it.

Reply:
I do like it